controllers/volsync/vshandler_test.go

```diff
@@ -664,6 +664,14 @@ var _ = Describe("VolSync_Handler", func() {
 					Expect(k8sClient.Create(ctx, dummyPSKSecret)).To(Succeed())
 					Expect(dummyPSKSecret.GetName()).NotTo(BeEmpty())
 
+					// Make sure the secret is created to avoid any timing issues
+					Eventually(func() error {
+						return k8sClient.Get(ctx, types.NamespacedName{
+							Name:      dummyPSKSecret.GetName(),
+							Namespace: dummyPSKSecret.GetNamespace(),
+						}, dummyPSKSecret)
+					}, maxWait, interval).Should(Succeed())
```

Comment:
So the test fails randomly since we did not wait until the secret was created in the testenv api server?

Reply:
That is my suggestion. In this file there are several places, we are creating the secrets, and after every such creation there is an 'Eventually' block. Only here there was no 'Eventually'.

Reply:
The next step is creating a little helper to create a secret and wait until it exists instead of duplicating the code.